src/components/apps/listing/Listing.js

```diff
@@ -0,0 +1,130 @@
+/**
+ * @author sriram
+ *
+ *  A component intended to be the parent to the apps's table view and
+ * thumbnail/tile view.
+ *
+ */
+import React, { useState } from "react";
+import { usePaginatedQuery } from "react-query";
+import { getApps } from "../../../serviceFacade/appServiceFacade";
+import TableView from "./TableView";
+import { TablePagination } from "@material-ui/core";
+
+function Listing(props) {
+    const [order, setOrder] = useState("asc");
+    const [orderBy, setOrderBy] = useState("name");
+    const [selected, setSelected] = useState([]);
+    const [lastSelectIndex, setLastSelectIndex] = useState(-1);
+    const [page, setPage] = useState(0);
+    const [rowsPerPage, setRowsPerPage] = useState(25);
+    const { status, resolvedData: data, error } = usePaginatedQuery(
+        ["getApps", { rowsPerPage, orderBy, order, page }],
+        getApps
+    );
```

Comment:
We need to make a choice here ...With `usePaginatedQuery` when user switches between pages, there won't be a loading skeleton. Instead the page will continue to show the current data until new data is available. If we want loading skeleton, then we need to switch back to `useQuery`. I feel we need the loading skeleton otherwise the users are going be confused and will start to think nothing happened.

Reply:
I switched to using `useQuery`.